poolmgr/client/client.go

```diff
@@ -60,12 +84,49 @@ func (c *Client) GetServiceForFunction(metadata *fission.Metadata) (string, erro
 	return string(svcName), nil
 }
 
-func (c *Client) TapService(serviceUrl *url.URL) error {
-	url := c.poolmgrUrl + "/v1/tapService"
+func (c *Client) tapService() {
+	for {
+		req := <-c.requestChan
+		switch req.requestType {
+		case TOUCH:
+			c.tappedByUrl[req.serviceUrl] = true
+		case SYNC:
+			for u := range c.tappedByUrl {
+				c.tapUrlChan <- u
+			}
+			if len(c.tappedByUrl) > 0 {
+				log.Printf("Tapped %v services in batch", len(c.tappedByUrl))
+			}
+			c.tappedByUrl = make(map[string]bool)
+		}
+	}
+}
+
+func (c *Client) tapWorker() {
+	for {
+		u := <-c.tapUrlChan
+		c._tapService(u)
+	}
+}
+
+func (c *Client) timer() {
```

Comment:
Thanks. I was worried about if there are too many requests and then the ticker channel may not be selected. I ran an experiment and it worked fine. Will make the changes you proposed.